driver-core/src/main/com/mongodb/client/model/expressions/Expressions.java

```diff
@@ -207,11 +207,24 @@ public static <R extends DocumentExpression> R current() {
                 .assertImplementsAllExpressions();
     }
 
+    /**
+     * Returns the "current" value as a {@linkplain MapExpression map} value.
+     * The "current" value is the top-level document currently being processed
+     * in the aggregation pipeline stage.
+     *
+     * @return the resulting value
+     * @param <R> the type of the resulting value
+     */
+    public static <R extends DocumentExpression> MapExpression<R> currentAsMap() {
+        return new MqlExpression<>((cr) -> new AstPlaceholder(new BsonString("$$CURRENT")))
+                .assertImplementsAllExpressions();
+    }
+
```

Comment:
I think this method should be generic over `<T extends Expression>` and return `MapExpression<T>`, where `T` is the type of map values (this last part affects the documentation).

Reply:
Right, this is an error.